File: dart2/CSharp/Dart2LexerBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using Antlr4.Runtime;

public abstract class Dart2LexerBase : Lexer
{
	protected Dart2LexerBase(ICharStream input)
		: base(input)
	{
	}

	public Dart2LexerBase(ICharStream input, TextWriter output, TextWriter errorOutput)
		: base(input, output, errorOutput) { }

	protected bool CheckNotOpenBrace()
	{
		return this.InputStream.LA(1) != '{';
	}
}

File: dart2/Cpp/Dart2LexerBase.cpp
#include "antlr4-runtime.h"
#include "Dart2LexerBase.h"
#include "Dart2Lexer.h"

Dart2LexerBase::Dart2LexerBase(antlr4::CharStream * input) : antlr4::Lexer(input)
{
    _input = input;
}

bool Dart2LexerBase::CheckNotOpenBrace()
{
    return this->_input->LA(1) != '{';
}
